File: Problem_1 Two_Sum.py
# // Time Complexity : O(n)
# // Space Complexity : O(n)
# // Did this code successfully run on Leetcode : Yes
# // Any problem you faced while coding this : No
#
#
# // Your code here along with comments explaining your approach

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        answer = []
        for i in range(len(nums)):
            num2 = target - nums[i]
            if (num2 in hashmap.keys()):
                num2_index = nums.index(num2)
                if (i != num2_index):
                    return sorted([i, num2_index])
            hashmap.update({nums[i]: i})

File: Problem1_TwoSum.java
/*
 * Time complexity: O(N) have to traverse array at least once.
 * Space Complexity: O(N) Worst case -takes all elements from array
 * 
 * */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> diffMap = new HashMap();
        
        for(int i=0; i<nums.length; i++){
            if(diffMap.containsKey(target-nums[i])){
                return new int[] {diffMap.get(target-nums[i]), i};
            }else{
                diffMap.put(target-nums[i],i);
                diffMap.put(nums[i],i); 
            }
        }
        
        return new int[] {-1,-1};
    }
}